### PR TITLE
Add userContext field to WebDriver BiDi's setPermission

### LINKED ISSUE #439: WebDriver BiDi: support user contexts in setPermission
User contexts in WebDriver BiDi represents the same concepts as the user agents in the infra spec: https://w3c.github.io/webdriver-bidi/#user-contexts According to the infra definition a user agent can be a user profile or a private browsing window.

There should be a way for a BiDi client to express which user agent should get the permissions.

### PR DESCRIPTION
**Pull request by OrKoN**
User contexts in WebDriver BiDi represents the same concepts as the user agents in the infra spec: https://w3c.github.io/webdriver-bidi/#user-contexts According to the infra definition a user agent can be a user profile or a private browsing window.

This PR adds an ability for the automation client to configure which "user agent" should the permissions apply too.

closes #439

The following tasks have been completed:

 * [x] Modified Web platform tests ([link](https://github.com/web-platform-tests/wpt/pull/44327))

Implementation commitment:

 * [ ] [WebKit](https://bugs.webkit.org/show_bug.cgi?id=265621)
 * [x] Blink (link to issue): https://github.com/GoogleChromeLabs/chromium-bidi/issues/1585
 * [ ] Gecko (link to issue): https://github.com/mozilla/standards-positions/issues/930

***
<a href="https://pr-preview.s3.amazonaws.com/OrKoN/permissions/pull/438.html" title="Last updated on Mar 8, 2024, 6:56 AM UTC (cf691dd)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/permissions/438/75ef57e...OrKoN:cf691dd.html" title="Last updated on Mar 8, 2024, 6:56 AM UTC (cf691dd)">Diff</a>

### CODE REVIEW

**Comment by OrKoN** [comment]
@miketaylr could you please take a look? Does the definition like this make sense from the perspective of the permissions spec? It looks like until now the spec didn't deal with multiple user agents at once.

File: index.html [hunk 1]
```diff
@@ -1459,6 +1461,9 @@ <h6 id="webdriver-bidi-command-permissions-setPermission">
                   </li>
                   <li>Let |state| be the value of the `state` field of |command parameters|.
                   </li>
+                  <li>Let |user context id| be the value of the `userContext` field of |command
```
Comment (miketaylr) [inline index.html:1464]: Trying to understand what a "user context" looks like... in https://w3c.github.io/webdriver-bidi/#user-context, it seems like a data structure that has an ID, has a storage partition, and has a collection of traversables? Is that right? Or is it strictly just a unique identifier and the rest happens by magic?

**Review by miketaylr — COMMENTED** [review]
(apologies for the delay here!)

File: index.html (hunk 1 above)
Comment (OrKoN) [reply index.html:1464]: The user context roughly corresponds to the user agent of the infra spec definition (except that the WebDriver can identify them). User context IDs are unique, except for the default user context (identified as "default"). In implementations, a user context is an incognito window in Chrome or a container in Firefox.

Comment (OrKoN) [reply index.html:1464]: @miketaylr is there anything else I could try to clarify?

**Comment by OrKoN** [comment]
Ah, sorry, it got auto-closed since I moved on with the vendor-prefixed implementation for now.

File: index.html (hunk 1 above)
Comment (OrKoN) [reply index.html:1464]: Sorry, closing was not intended.

**Review by miketaylr — APPROVED** [review]
This LGTM, mind taking a look @marcoscaceres?

@OrKoN - do we have interest from another vendor to implement this?

**Comment by OrKoN** [comment]
I have not received any feedback from other vendors on this particular change (except for the general implementation interest for WebDriver BiDi/permissions).

**Review by marcoscaceres — APPROVED** [review]
This seems fine.

**Comment by miketaylr** [comment]
```
  Error: ROR] Couldn't find "session" in this document or other cited documents: [dom], [fetch], [html], [infra], [permissions-policy], [url], [w3c-process], [webdriver], and [webidl].
```

**Comment by miketaylr** [comment]
@OrKoN mind rebasing? Then we can merge.

**Comment by OrKoN** [comment]
@miketaylr @marcoscaceres thanks for reviewing, I have re-based.

**Pull request merged**